### PR TITLE
[MODEXPS-251] - Delete sensitive information from the log

### PR DESCRIPTION
**Pull request by Dmitriy-Butramyou**
## Purpose
[MODEXPS-251](https://issues.folio.org/browse/MODEXPS-251) - Delete sensitive information from the log

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] Check logging
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Comment by Dmitriy-Butramyou** [comment]
image

**Review by Abdulkhakimov — APPROVED** [review]

File: src/main/java/org/folio/des/util/LogMaskingConverter.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+  @Override
+  public void format(LogEvent event, StringBuilder toAppendTo) {
+    String messageString = toAppendTo.toString();
+    toAppendTo.delete(0, toAppendTo.length());
+    String maskedMessage;
+    try {
+      maskedMessage = mask(messageString);
+    } catch (Exception e) {
+      log.error("Failed While Masking");
```
Comment (SerhiiNosko) [inline src/main/java/org/folio/des/util/LogMaskingConverter.java:44]: better to add e or e.getMessage()

File: src/main/java/org/folio/des/util/LogMaskingConverter.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    toAppendTo.append(maskedMessage);
+  }
+
+  private String mask(String message) {
+    Matcher matcher;
+    StringBuffer buffer = new StringBuffer();
+
+    matcher = SERVER_ADDRESS_PATTERN.matcher(message);
+    maskMatcher(matcher, buffer, SERVER_ADDRESS_PATTERN_REPLACEMENT);
+    message=buffer.toString();
```
Comment (SerhiiNosko) [inline src/main/java/org/folio/des/util/LogMaskingConverter.java:56]: code formatting issue here, add spaces before and after =

Comment (Dmitriy-Butramyou) [reply src/main/java/org/folio/des/util/LogMaskingConverter.java]: refactored

File: src/main/java/org/folio/des/util/LogMaskingConverter.java (hunk 1 above)
Comment (Dmitriy-Butramyou) [reply src/main/java/org/folio/des/util/LogMaskingConverter.java]: added

**Review by azizbekxm — APPROVED** [review]

**Pull request merged**